nexus/src/app/sagas/instance_create.rs

```diff
@@ -1077,8 +1076,12 @@ async fn sic_set_boot_disk(
         .await
         .map_err(ActionError::action_failed)?;
 
-    let initial_configuration =
-        nexus_db_model::InstanceUpdate { boot_disk_id: Some(authz_disk.id()) };
+    let initial_configuration = nexus_db_model::InstanceUpdate {
+        boot_disk_id: Some(authz_disk.id()),
+        // If this is `None`, whatever previous value was set when creating the
+        // instance will be unset and replaced with `NULL`. So don't do that!
```

Comment:
oh, there probably should be a way to go from an `Instance` to a no-op `InstanceUpdate` so this doesn't become awful. it looks like here we'd be able to get the created instance by `sagactx.lookup::<db::model::Instance>("instance_record")`?

sorry, i should have thought of that earlier

Reply:
Huh, I'm surprised to see this comment--I thought we only set `treat_none_as_null` for `InstanceRuntimeState` and not for `Instance`. Does that setting get pulled up to the main `Instance` struct by the `#[diesel(embed)]`? (Am I just not grokking how it works?)

Reply:
Oh, we're setting that attribute on `InstanceUpdate`. Withdrawn!

Reply:
`InstanceUpdate` specifically is _also_ [`treat_none_as_null`](https://github.com/oxidecomputer/omicron/blob/feff3c5c76147ffc319471c41b8bf7d60df3d56e/nexus/db-model/src/instance.rs#L533), no relation to `Instance` or `#[diesel(embed)]`.